cmd/sonarExecuteScan.go

```diff
@@ -0,0 +1,242 @@
+		sonar.Options = append(sonar.Options, "sonar.projectVersion="+options.ProjectVersion)
+	}
+
+	handlePullRequest(options)
+
+	loadSonarScanner(options.SonarScannerDownloadURL, client)
+
+	loadCertificates(runner, "", client)
+
+	log.Entry().
+		WithField("command", sonar.Binary).
+		WithField("options", sonar.Options).
+		WithField("environment", sonar.Environment).
+		Debug("Executing sonar scan command")
+
+	runner.SetEnv(sonar.Environment)
+
+	sonar.Options = SliceUtils.Prefix(sonar.Options, "-D")
+	if err := runner.RunExecutable(sonar.Binary, sonar.Options...); err != nil {
+		log.Entry().WithError(err).Fatal("Failed to execute scan command")
+	}
+}
+
+func handlePullRequest(options sonarExecuteScanOptions) {
+	if len(options.ChangeID) > 0 {
+		if options.LegacyPRHandling {
+			// see https://docs.sonarqube.org/display/PLUG/GitHub+Plugin
+			sonar.Options = append(sonar.Options, "sonar.analysis.mode=preview")
+			sonar.Options = append(sonar.Options, "sonar.github.pullRequest="+options.ChangeID)
+			if len(options.GithubAPIURL) > 0 {
+				sonar.Options = append(sonar.Options, "sonar.github.endpoint="+options.GithubAPIURL)
+			}
+			if len(options.GithubToken) > 0 {
+				sonar.Options = append(sonar.Options, "sonar.github.oauth="+options.GithubToken)
+			}
+			if len(options.Owner) > 0 && len(options.Repository) > 0 {
+				sonar.Options = append(sonar.Options, "sonar.github.repository="+options.Owner+"/"+options.Repository)
+			}
+			if options.DisableInlineComments {
+				sonar.Options = append(sonar.Options, "sonar.github.disableInlineComments="+strconv.FormatBool(options.DisableInlineComments))
+			}
+		} else {
+			// see https://sonarcloud.io/documentation/analysis/pull-request/
+			sonar.Options = append(sonar.Options, "sonar.pullrequest.key="+options.ChangeID)
+			sonar.Options = append(sonar.Options, "sonar.pullrequest.base="+options.ChangeTarget)
+			sonar.Options = append(sonar.Options, "sonar.pullrequest.branch="+options.ChangeBranch)
+			sonar.Options = append(sonar.Options, "sonar.pullrequest.provider="+options.PullRequestProvider)
+			//TODO: use toLowerCase ?
+			if options.PullRequestProvider == "GitHub" {
+				sonar.Options = append(sonar.Options, "sonar.pullrequest.github.repository="+options.Owner+"/"+options.Repository)
+			} else {
+				log.Entry().Fatal("Pull-Request provider '" + options.PullRequestProvider + "' is not supported!")
+			}
+		}
+	}
+}
+
+func loadSonarScanner(url string, client piperhttp.Downloader) {
+	if scannerPath, err := execLookPath(sonar.Binary); err == nil {
+		// using existing sonar-scanner
+		log.Entry().WithField("path", scannerPath).Debug("Using local Sonar scanner cli")
+	} else {
+		// download sonar-scanner-cli from url to .sonar-scanner/
+		log.Entry().WithField("url", url).Debug("Downloading Sonar scanner cli")
+		if len(url) == 0 {
+			log.Entry().Error("Download url for Sonar scanner cli missing")
+		}
+		// download sonar-scanner-cli into TEMP folder
+		tmpFolder := getTempDir()
```

Comment:
not sure if this is a good idea, e.g. on a Jenkins ...
We might be in a file area where the current user does not have permissions.

Shouldn't we rather stay in the working directory here?

Reply:
Would `ioutil.TempDir(".", "temp-")` be ok?

Reply:
> We might be in a file area where the current user does not have permissions.

I would expect the implementation to handle this, e.g. throw error.

Reply:
I guess this would be ok